main/src/main/java/cgeo/geocaching/unifiedmap/UnifiedMapActivity.java

```diff
@@ -856,16 +855,6 @@ public boolean onOptionsItemSelected(@NonNull final MenuItem item) {
             mapFragment.selectThemeOptions(this);
         } else if (id == R.id.menu_routetrack) {
             routeTrackUtils.showPopup(viewModel.individualRoute.getValue(), viewModel::setTarget);
-        } else if (id == R.id.menu_select_mapview) {
```

Comment:
Have you cross-checked whether creating the menu entries statically in `onPrepareOptionsMenu` recognizes map source changes (eg. deleting one of the offline maps)?

Reply:
Oh, so that's the idea behind that strange code (which btw exists only in Unified, not in legacy map). For adding new maps this wasn't a problem as the map is added from another activity, but with the re-added Delete it can indeed cause problems.
However I'm wondering whether this relatively rare case is worth the hassle of the working currently implemented (again only for Unified), recreating the menu contents on the fly...